apps/native-component-list/src/screens/StoreReview.tsx

```diff
@@ -1,71 +1,61 @@
+// import { StackNavigationProp } from '@react-navigation/stack';
+import * as StoreReview from 'expo-store-review';
 import React from 'react';
 import { StyleSheet, Text, View } from 'react-native';
-import * as StoreReview from 'expo-store-review';
 
 import Button from '../components/Button';
-import CannyFooter from '../components/CannyFooter';
 import Colors from '../constants/Colors';
 
-class StoreReviewScreen extends React.Component {
-  static navigationOptions = {
-    title: 'Store Review',
-  };
-
-  state = {
-    isAvailable: false,
-  };
+type Props = {
+  navigation: any; // StackNavigationProp<any>;
+};
 
-  async componentDidMount() {
-    const isAvailable = await StoreReview.isAvailableAsync();
-    this.setState({
-      isAvailable,
-    });
+function getStoreUrlInfo(): string {
+  const storeUrl = StoreReview.storeUrl();
+  if (storeUrl) {
+    return `On iOS <10.3, or Android devices pressing this button will open ${storeUrl}.`;
   }
+  return 'You will need to add ios.appStoreUrl, and android.playStoreUrl to your app.config.js in order to use this feature on Android.';
+}
 
-  onRequestReview = () => StoreReview.requestReview();
+function StoreReviewScreen({ navigation }: Props) {
+  // React.useLayoutEffect(() => {
+  //   navigation.setOptions({
+  //     title: 'Store Review',
+  //   });
+  // }, [navigation]);
 
-  get isSupportedText() {
-    return this.state.isAvailable ? 'is available :D' : 'is not available! D:';
-  }
+  const [isAvailable, setAvailable] = React.useState<boolean>(false);
 
-  get storeUrl() {
-    const storeUrl = StoreReview.storeUrl();
-    if (storeUrl) {
-      return `On iOS <10.3, or Android devices pressing this button will open ${storeUrl}.`;
-    } else {
-      return 'You will need to add ios.appStoreUrl, and android.playStoreUrl to your app.json in order to use this feature.';
-    }
-  }
-
-  render() {
-    return (
-      <View style={styles.container}>
-        <View>
-          <View style={styles.textContainer}>
-            <Text style={styles.isSupportedText}>Native Store Review {this.isSupportedText}</Text>
-            <Text style={styles.storeUrlText}>{this.storeUrl}</Text>
-          </View>
+  React.useEffect(() => {
+    StoreReview.isAvailableAsync().then(setAvailable);
```

Comment:
Shouldn't need `<boolean>` here, the inferred type should be sufficient (and in TS we generally use inference)